nemo/core/classes/modelPT.py

```diff
@@ -80,32 +80,23 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
         self._trainer = trainer
 
         if self._cfg is not None:  # TODO: This check is redundant since we know cfg is an instance of DictConfig
-            if 'train_ds' in self._cfg and self._cfg.train_ds is not None:
-                self.setup_training_data(self._cfg.train_ds)
-
-            if 'validation_ds' in self._cfg and self._cfg.validation_ds is not None:
-                # Set some placeholder overriden by helper method
-                self._validation_loss_idx = 0
-                self._validation_names = None
-                self._validation_dl = None  # type: torch.utils.data.DataLoader
+            try:
+                if 'train_ds' in self._cfg and self._cfg.train_ds is not None:
+                    self.setup_training_data(self._cfg.train_ds)
 
-                model_utils.resolve_validation_dataloaders(model=self)
+            except Exception:
```

Comment:
Can we stop catching `Exceptions`? Catch the `Exception` you expect to occur, otherwise just let it fail.
If you want to print a helper message, then feel free to print the helper function, but then reraise the old Exception.

Reply:
There are upto 7 (as far as far as I know) messages that can occur here, cause this is dependent on several subclass code behaviour, modalities of operation - single vs multiple data loaders, different ways of execution - fadtvpath vs explicit path resolution, different domains handling the Dataloaders differently. We also do not want to raise the exception as then we cannot resume model loading. 

At worst, I can look into catching the exception stack trace and logging it into a debug level, but catching explicitly each exception is not reasonable here.

Reply:
I'm fact without this patch, restore from .Nemo files is basically impossible on different systems than the one that was used to generate the .Nemo file.

Reply:
Ok, let's simply just print the original exception as well and then we can merge.

Reply:
Done

Reply:
I might even think we should move it up from debug to warning, so that user's can see the error to have an indication what in their config they should fix.